homeassistant/components/inels/config_flow.py

```diff
@@ -0,0 +1,273 @@
+                        CONF_PASSWORD: user_input.get(CONF_PASSWORD),
+                        MQTT_TRANSPORT: user_input.get(MQTT_TRANSPORT),
+                        CONF_DISCOVERY: True,
+                    },
+                )
+
+            errors["base"] = connect_val_to_error(test_connect)
+        else:
+            user_input = {}
+
+        fields = OrderedDict()
+        fields[vol.Required(CONF_HOST, default=user_input.get(CONF_HOST))] = str
+        fields[
+            vol.Required(
+                CONF_PORT,
+                default=1883
+                if user_input.get(CONF_PORT) is None
+                else user_input.get(CONF_PORT),
+            )
+        ] = vol.Coerce(int)
+        fields[vol.Optional(CONF_USERNAME, default=user_input.get(CONF_USERNAME))] = str
+        fields[vol.Optional(CONF_PASSWORD, default=user_input.get(CONF_PASSWORD))] = str
+        fields[vol.Required(MQTT_TRANSPORT, default="tcp")] = vol.In(
+            ["tcp", "websockets"]
+        )
+
+        return self.async_show_form(
+            step_id="setup", data_schema=vol.Schema(fields), errors=errors
+        )
+
+    async def async_step_hassio(self, discovery_info: HassioServiceInfo) -> FlowResult:
+        """Receive a Hass.io discovery."""
+        await self._async_handle_discovery_without_unique_id()
+        self._hassio_discovery = discovery_info.config
+
+        return await self.async_step_confirm()
+
+    async def async_step_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Confirm a discovery."""
+        errors = {}
+        assert self._hassio_discovery
+
+        if user_input is not None:
+            data = self._hassio_discovery
+            test_connect = await self.hass.async_add_executor_job(
+                try_connection,
+                self.hass,
+                data[CONF_HOST],
+                data[CONF_PORT],
+                data.get(CONF_USERNAME),
+                data.get(CONF_PASSWORD),
+                data.get(MQTT_TRANSPORT),
+            )
+
+            if test_connect is None:
+                return self.async_create_entry(
+                    title=TITLE,
+                    data={
+                        CONF_HOST: data[CONF_HOST],
+                        CONF_PORT: data[CONF_PORT],
+                        CONF_USERNAME: data.get(CONF_USERNAME),
+                        CONF_PASSWORD: data.get(CONF_PASSWORD),
+                        MQTT_TRANSPORT: data.get(MQTT_TRANSPORT),
+                        CONF_DISCOVERY: True,
+                    },
+                )
+
+            errors["base"] = connect_val_to_error(test_connect)
+
+        return self.async_show_form(
+            step_id="confirm",
+            description_placeholders={"addon": self._hassio_discovery["addon"]},
+            errors=errors,
+        )
+
+
+class InelsOptionsFlowHandler(config_entries.OptionsFlow):
```

Comment:
I'd remove the options flow for the initial PR.

Maintainers really like the project to be as small as possible, and this isn't essential, so you could always add it back in a small PR after this ones gets merged

Reply:
I removed it to make it more simple